rudder-core/src/main/scala/com/normation/rudder/reports/ReportingConfiguration.scala

```diff
@@ -107,5 +130,32 @@ class AgentRunIntervalServiceImpl (
       )
     }
   }
+
+  override def getNodeReportingConfigurations(nodeIds: Set[NodeId]): Box[Map[NodeId, ResolvedAgentRunInterval]] = {
+    for {
+      gInterval  <- readGlobalInterval()
+      gHeartbeat <- readGlobalHeartbeat()
+      nodeInfos  <- nodeInfoService.getAll()
+    } yield {
+      nodeIds.map { nodeId =>
+
+        if(nodeId == Constants.ROOT_POLICY_SERVER_ID) {
```

Comment:
I'm not sure it's what is done on the promise side: https://github.com/Normation/rudder/blob/branches/rudder/3.0/rudder-core/src/main/scala/com/normation/rudder/services/policies/SystemVariableService.scala#L285

if(nodeInfo.isPolicyServer) {....